tests/core/pyspec/eth2spec/test/eip7732/epoch_processing/test_process_builder_pending_payments.py

```diff
@@ -0,0 +1,281 @@
+    # Advance past genesis epochs
+    next_epoch(spec, state)
+    next_epoch(spec, state)
+
+    # Fill both epochs of the queue with test data
+    test_weight = 12345
+    for i in range(2 * spec.SLOTS_PER_EPOCH):
+        payment = create_builder_pending_payment(spec, i, spec.MIN_ACTIVATION_BALANCE, test_weight)
+        state.builder_pending_payments[i] = payment
+
+    # Store the second epoch data for comparison
+    second_epoch_payments = []
+    for i in range(spec.SLOTS_PER_EPOCH, 2 * spec.SLOTS_PER_EPOCH):
+        payment = state.builder_pending_payments[i]
+        second_epoch_payments.append((payment.weight, payment.withdrawal.builder_index))
+
+    yield from run_epoch_processing_with(spec, state, "process_builder_pending_payments")
+
+    # First SLOTS_PER_EPOCH should be the old second epoch
+    for i in range(spec.SLOTS_PER_EPOCH):
+        payment = state.builder_pending_payments[i]
+        expected_weight, expected_builder_index = second_epoch_payments[i]
+        assert payment.weight == expected_weight
+        assert payment.withdrawal.builder_index == expected_builder_index
+
+    # Second SLOTS_PER_EPOCH should be empty
+    for i in range(spec.SLOTS_PER_EPOCH, 2 * spec.SLOTS_PER_EPOCH):
+        payment = state.builder_pending_payments[i]
+        assert payment.weight == 0
+        assert payment.withdrawal.amount == 0
+        assert payment.withdrawal.builder_index == 0
+
+
+@with_eip7732_and_later
+@spec_state_test
+def test_process_builder_pending_payments_large_amount_churn_impact(spec, state):
+    """Test that large payment amounts impact the exit churn correctly."""
+    # Advance past genesis epochs
+    next_epoch(spec, state)
+    next_epoch(spec, state)
+
+    builder_index = 0
+    large_amount = spec.MAX_EFFECTIVE_BALANCE_ELECTRA
+    quorum = get_quorum_threshold(spec, state)
+    weight = quorum + 1
+
+    payment = create_builder_pending_payment(spec, builder_index, large_amount, weight)
+    state.builder_pending_payments[0] = payment
+
+    pre_exit_balance_to_consume = state.exit_balance_to_consume
+    pre_earliest_exit_epoch = state.earliest_exit_epoch
+
+    yield from run_epoch_processing_with(spec, state, "process_builder_pending_payments")
+
+    # Should have one withdrawal
+    assert len(state.builder_pending_withdrawals) >= 1
+    withdrawal = state.builder_pending_withdrawals[len(state.builder_pending_withdrawals) - 1]
+    assert withdrawal.amount == large_amount
+
+    # Exit queue should be impacted by large amount processing
+    # Check that either balance was consumed or exit epoch was impacted
+    assert (
+        state.exit_balance_to_consume != pre_exit_balance_to_consume
+        or state.earliest_exit_epoch != pre_earliest_exit_epoch
+    )
+
+    # Withdrawable epoch should account for the large amount
+    current_epoch = spec.get_current_epoch(state)
+    assert withdrawal.withdrawable_epoch > current_epoch
```

Comment:
This would happen also without a large churn: we need to actually check that the churn was updated to the right value.

Reply:
added in L282

Reply:
I don't see a L282 here